src/lib.rs

```diff
@@ -1334,22 +1334,23 @@ impl Node {
 			}
 			Err(payment::PaymentError::Sending(e)) => {
 				log_error!(self.logger, "Failed to send payment: {:?}", e);
-
-				let payment = PaymentDetails {
-					preimage: None,
-					hash: payment_hash,
-					secret: payment_secret,
-					amount_msat: invoice.amount_milli_satoshis(),
-					direction: PaymentDirection::Outbound,
-					status: PaymentStatus::SendingFailed,
-				};
-				self.payment_store.insert(payment)?;
-
 				match e {
 					channelmanager::RetryableSendFailure::DuplicatePayment => {
 						Err(Error::DuplicatePayment)
 					}
-					_ => Err(Error::PaymentSendingFailed),
+					_ => {
+						let payment = PaymentDetails {
+							preimage: None,
+							hash: payment_hash,
+							secret: payment_secret,
+							amount_msat: invoice.amount_milli_satoshis(),
+							direction: PaymentDirection::Outbound,
+							status: PaymentStatus::SendingFailed,
+						};
+
+						self.payment_store.insert(payment)?;
```

Comment:
So we have any tests that check the payment store? Might be worth adding a test case for duplicate payments.

Reply:
But we don't check that the status remains the same as before in case of a duplicate error, IIUC.